File: ecommerce-dashboard/app/static/js/ebay.js
"use strict";

/* ── eBay Legacy Tab ── */

/* ── Load ── */
async function loadEbay() {
  const [summaryRes, ordersRes] = await Promise.all([
    fetchJson(`${API_BASE}/ebay/summary`),
    fetchJson(`${API_BASE}/ebay/orders`),
  ]);
  state.ebaySummary = summaryRes;
  state.ebayOrders = ordersRes.orders || [];
  state.ebayOrdersTotal = ordersRes.total || 0;
}

/* ── Category labels ── */
const EBAY_CATEGORY_LABELS = {
  order: "Bestellung",
  computer: "Computer",
  return: "Ruecksendung",
};

/* ── Render ── */
function renderEbay() {
  const summary = state.ebaySummary || {};
  const kpis = summary.kpis || {};
  const shops = summary.shops || [];
  const topArticles = summary.top_articles || [];
  const importMeta = summary.import_meta || {};

  /* KPIs */
  if (els.ebayKpiOrders) {
    els.ebayKpiOrders.textContent = NUMBER_FMT.format(kpis.total_orders || 0);
  }
  if (els.ebayKpiOrdersSub) {
    const returns = kpis.total_returns || 0;
    const dateRange = kpis.first_date && kpis.last_date
      ? `${formatDateTokenLabel(kpis.first_date)} - ${formatDateTokenLabel(kpis.last_date)}`
      : "";
    els.ebayKpiOrdersSub.textContent = returns > 0
      ? `${returns} Ruecksendungen | ${dateRange}`
      : dateRange || "Legacy eBay Daten";
  }
  if (els.ebayKpiRevenue) {
    els.ebayKpiRevenue.textContent = MONEY_FMT.format(kpis.total_revenue || 0);
  }
  if (els.ebayKpiCosts) {
    const costs = (kpis.total_purchase || 0) + (kpis.total_fees || 0);
    els.ebayKpiCosts.textContent = MONEY_FMT.format(costs);
  }
  if (els.ebayKpiCostsSub) {
    els.ebayKpiCostsSub.textContent = `Einkauf ${MONEY_FMT.format(kpis.total_purchase || 0)} + Gebuehren ${MONEY_FMT.format(kpis.total_fees || 0)}`;
  }
  if (els.ebayKpiProfit) {
    els.ebayKpiProfit.textContent = MONEY_FMT.format(kpis.total_profit || 0);
  }
  if (els.ebayKpiProfitSub) {
    els.ebayKpiProfitSub.textContent = `Marge: ${NUMBER_FMT.format(kpis.margin_pct || 0)}%`;
  }

  /* Shops table */
  if (els.ebayShopsBody) {
    els.ebayShopsBody.innerHTML = shops.length
      ? shops.map(s => {
          const dateRange = s.first_date && s.last_date
            ? `${formatDateTokenLabel(s.first_date)} - ${formatDateTokenLabel(s.last_date)}`
            : "-";
          return `<tr>
            <td><strong>${escapeHtml(s.shop)}</strong></td>
            <td>${s.count}</td>
            <td>${escapeHtml(dateRange)}</td>
            <td>${MONEY_FMT.format(s.revenue)}</td>
            <td>${MONEY_FMT.format(s.fees)}</td>
            <td>${MONEY_FMT.format(s.purchase)}</td>
            <td class="${s.profit >= 0 ? 'profit-positive' : 'profit-negative'}">${MONEY_FMT.format(s.profit)}</td>
          </tr>`;
        }).join("")
      : '<tr><td colspan="7">Keine eBay Daten importiert.</td></tr>';
  }
  if (els.ebayShopsMeta) {
    els.ebayShopsMeta.textContent = `${shops.length} Shop${shops.length !== 1 ? "s" : ""}`;
  }

  /* Populate shop filter dropdown */
  if (els.ebayShopFilter) {
    const currentVal = els.ebayShopFilter.value;
    const shopNames = [...new Set((state.ebayOrders || []).map(o => o.shop))].sort();
    let optionsHtml = '<option value="">Alle Shops</option>';
    for (const name of shopNames) {
      const sel = name === currentVal ? " selected" : "";
      optionsHtml += `<option value="${escapeHtml(name)}"${sel}>${escapeHtml(name)}</option>`;
    }
    els.ebayShopFilter.innerHTML = optionsHtml;
  }

  /* Orders table (filtered) */
  renderEbayOrders();

  /* Top articles */
  if (els.ebayTopArticlesBody) {
    els.ebayTopArticlesBody.innerHTML = topArticles.length
      ? topArticles.map(a => `<tr>
          <td>${escapeHtml(a.artikel || "-")}</td>
          <td>${a.count}</td>
          <td>${MONEY_FMT.format(a.revenue)}</td>
          <td class="${a.profit >= 0 ? 'profit-positive' : 'profit-negative'}">${MONEY_FMT.format(a.profit)}</td>
        </tr>`).join("")
      : '<tr><td colspan="4">Keine Daten.</td></tr>';
  }

  /* Import info */
  if (els.ebayImportInfo) {
    if (importMeta && importMeta.imported_at) {
      els.ebayImportInfo.innerHTML = [
        `Quelle: <strong>${escapeHtml(importMeta.source_file || "-")}</strong>`,
        `Importiert: <strong>${formatDate(importMeta.imported_at)}</strong>`,
        `Shops: <strong>${escapeHtml(importMeta.shops || "-")}</strong>`,
        `Bestellungen: <strong>${importMeta.total_orders || 0}</strong> | Ruecksendungen: <strong>${importMeta.total_returns || 0}</strong>`,
        '<span style="opacity:.6">Hinweis: eBay Legacy-Daten — kein aktiver Verkauf, nur Dokumentation.</span>',
      ].join("<br>");
    } else {
      els.ebayImportInfo.textContent = "Keine eBay Daten importiert. Bitte import_ebay.py ausfuehren.";
    }
  }
}

function renderEbayOrders() {
  const allOrders = state.ebayOrders || [];
  const shopFilter = els.ebayShopFilter ? els.ebayShopFilter.value : "";
  const categoryFilter = els.ebayCategoryFilter ? els.ebayCategoryFilter.value : "";

  const filtered = allOrders.filter(o => {
    if (shopFilter && o.shop !== shopFilter) return false;
    if (categoryFilter && o.category !== categoryFilter) return false;
    return true;
  });

  if (els.ebayOrdersMeta) {
    els.ebayOrdersMeta.textContent = `${filtered.length} Zeilen`;
  }

  if (els.ebayOrdersBody) {
    els.ebayOrdersBody.innerHTML = filtered.length
      ? filtered.map(o => {
          const isReturn = o.is_return === 1;
          const rowClass = isReturn ? ' class="return-row"' : "";
          const catLabel = EBAY_CATEGORY_LABELS[o.category] || o.category || "-";
          const gewinn = o.gewinn != null ? o.gewinn : 0;
          return `<tr${rowClass}>
            <td data-label="Datum">${o.datum ? formatDateTokenLabel(o.datum) : "-"}</td>
            <td data-label="Shop">${escapeHtml(o.shop || "-")}</td>
            <td data-label="Kategorie"><span class="badge badge-${isReturn ? 'cancel' : o.category === 'computer' ? 'partial' : 'ok'}">${escapeHtml(catLabel)}</span></td>
            <td data-label="Artikel">${escapeHtml(o.artikel || "-")}</td>
            <td data-label="Kunde">${escapeHtml(o.kunde_name || "-")}</td>
            <td data-label="Order Nr.">${escapeHtml(o.order_number || "-")}</td>
            <td data-label="Preis">${o.preis != null ? MONEY_FMT.format(o.preis) : "-"}</td>
            <td data-label="Gebuehren">${MONEY_FMT.format(o.gebuehren || 0)}</td>
            <td data-label="Einkauf">${o.ali_preis != null ? MONEY_FMT.format(o.ali_preis) : "-"}</td>
            <td data-label="Gewinn" class="${gewinn >= 0 ? 'profit-positive' : 'profit-negative'}">${MONEY_FMT.format(gewinn)}</td>
          </tr>`;
        }).join("")
      : '<tr><td colspan="10">Keine eBay Bestellungen fuer aktuellen Filter.</td></tr>';
  }
}

/* formatDateTokenLabel is provided by core.js */

File: ecommerce-dashboard/app/services/ebay.py
"""eBay legacy data service — read-only access to imported eBay order data."""

from __future__ import annotations

import sqlite3
from typing import Any, Optional

from app.config import EBAY_DB_PATH


def _connect() -> Optional[sqlite3.Connection]:
    if not EBAY_DB_PATH.exists():
        return None
    conn = sqlite3.connect(EBAY_DB_PATH)
    conn.row_factory = sqlite3.Row
    return conn


def get_ebay_orders(
    *,
    shop: Optional[str] = None,
    category: Optional[str] = None,
    include_returns: bool = True,
) -> list[dict[str, Any]]:
    """Return all eBay orders, optionally filtered by shop/category."""
    conn = _connect()
    if conn is None:
        return []

    clauses: list[str] = []
    params: list[Any] = []

    if shop:
        clauses.append("shop = ?")
        params.append(shop)
    if category:
        clauses.append("category = ?")
        params.append(category)
    if not include_returns:
        clauses.append("is_return = 0")

    where = f"WHERE {' AND '.join(clauses)}" if clauses else ""

    rows = conn.execute(
        f"""
        SELECT id, shop, category, artikel, kunde_name, order_number,
               datum, preis, gebuehren, nach_gebuehren, ali_preis,
               provision_rate, gewinn, is_return
        FROM ebay_orders
        {where}
        ORDER BY datum ASC, id ASC
        """,
        params,
    ).fetchall()

    conn.close()
    return [dict(r) for r in rows]


def get_ebay_summary() -> dict[str, Any]:
    """Build KPI summary for the eBay tab."""
    conn = _connect()
    if conn is None:
        return {"available": False}

    # Overall totals (exclude returns from revenue/profit KPIs)
    totals = conn.execute(
        """
        SELECT
            COUNT(*)                          AS total_count,
            COALESCE(SUM(preis), 0)           AS total_revenue,
            COALESCE(SUM(gebuehren), 0)       AS total_fees,
            COALESCE(SUM(ali_preis), 0)       AS total_purchase,
            COALESCE(SUM(gewinn), 0)          AS total_profit,
            COALESCE(SUM(nach_gebuehren), 0)  AS total_after_fees,
            MIN(datum)                        AS first_date,
            MAX(datum)                        AS last_date
        FROM ebay_orders
        WHERE is_return = 0
        """
    ).fetchone()

    returns_row = conn.execute(
        "SELECT COUNT(*) AS cnt FROM ebay_orders WHERE is_return = 1"
    ).fetchone()

    # Per-shop breakdown
    shop_rows = conn.execute(
        """
        SELECT
            shop,
            COUNT(*)                          AS count,
            COALESCE(SUM(preis), 0)           AS revenue,
            COALESCE(SUM(gebuehren), 0)       AS fees,
            COALESCE(SUM(ali_preis), 0)       AS purchase,
            COALESCE(SUM(gewinn), 0)          AS profit,
            MIN(datum)                        AS first_date,
            MAX(datum)                        AS last_date
        FROM ebay_orders
        WHERE is_return = 0
        GROUP BY shop
        ORDER BY shop
        """
    ).fetchall()

    # Top articles by revenue
    top_articles = conn.execute(
        """
        SELECT
            artikel,
            COUNT(*)                    AS count,
            COALESCE(SUM(preis), 0)     AS revenue,
            COALESCE(SUM(gewinn), 0)    AS profit
        FROM ebay_orders
        WHERE is_return = 0
        GROUP BY artikel
        ORDER BY revenue DESC
        LIMIT 15
        """
    ).fetchall()

    # Import metadata
    meta_row = conn.execute(
        "SELECT * FROM ebay_import_meta WHERE id = 1"
    ).fetchone()

    conn.close()

    total_revenue = totals["total_revenue"] if totals else 0
    total_profit = totals["total_profit"] if totals else 0
    margin_pct = (total_profit / total_revenue * 100) if total_revenue else 0

    return {
        "available": True,
        "kpis": {
            "total_orders": totals["total_count"] if totals else 0,
            "total_returns": returns_row["cnt"] if returns_row else 0,
            "total_revenue": round(total_revenue, 2),
            "total_fees": round(totals["total_fees"], 2) if totals else 0,
            "total_after_fees": round(totals["total_after_fees"], 2) if totals else 0,
            "total_purchase": round(totals["total_purchase"], 2) if totals else 0,
            "total_profit": round(total_profit, 2),
            "margin_pct": round(margin_pct, 1),
            "first_date": totals["first_date"] if totals else None,
            "last_date": totals["last_date"] if totals else None,
        },
        "shops": [
            {
                "shop": r["shop"],
                "count": r["count"],
                "revenue": round(r["revenue"], 2),
                "fees": round(r["fees"], 2),
                "purchase": round(r["purchase"], 2),
                "profit": round(r["profit"], 2),
                "first_date": r["first_date"],
                "last_date": r["last_date"],
            }
            for r in shop_rows
        ],
        "top_articles": [
            {
                "artikel": r["artikel"],
                "count": r["count"],
                "revenue": round(r["revenue"], 2),
                "profit": round(r["profit"], 2),
            }
            for r in top_articles
        ],
        "import_meta": dict(meta_row) if meta_row else None,
    }
